src/core/server/saved_objects/routes/export.ts

```diff
@@ -0,0 +1,97 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { schema } from '@kbn/config-schema';
+import stringify from 'json-stable-stringify';
+import {
+  createPromiseFromStreams,
+  createMapStream,
+  createConcatStream,
+} from '../../../../legacy/utils/streams';
```

Comment:
The whole SO import/export APIs rely on our legacy stream utilities. As we don't want to loose the (not yet used but anyway) http streaming feature for the export and import, I kept using them (changing that would mean some structural changes in the `getSortedObjectsForExport` method and it's underlying components.  This is not the first time we are importing this in core SO code though. We probably will have to move it to core at some point (FTR services also uses them, so this will not be removed when we remove legacy, at least no without rewrite), but the PR is already big, did not want to do it here.